### PR TITLE
refactor(snap): edgex-snap-hooks related upgrade

### PR DESCRIPTION
**Pull request by MonicaisHer**
This PR upgrades snap hooks to [2.4.1 ](https://github.com/canonical/edgex-snap-hooks/releases/tag/v2.4.1)with new function and refactor:
- Combine install hook and configure hook
- Use library function for copying directory
- Remove export bash command to process valid .env files
- Use log package for logging
- Use library function for autostart processing

Signed-off-by: Mengyi Wang <mengyi.wang@canonical.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-rfid-llrp-inventory/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)  [app-rfid-llrp-inventory test suite](https://github.com/canonical/edgex-snap-testing/tree/main/test/suites/app-rfid-llrp-inventory) passed with `FULL_CONFIG_TEST=true`
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

1. Install and validate that all config files have been copied in right place:
```
tree /var/snap/edgex-app-rfid-llrp-inventory/current/config
.
/var/snap/edgex-app-rfid-llrp-inventory/current/config
└── app-rfid-llrp-inventory
    └── res
        └── configuration.toml

2 directories, 1 file
```
2. Setup and start:
```
sudo snap set edgex-app-rfid-llrp-inventory app-options=true
sudo snap set edgex-app-rfid-llrp-inventory config.writable-loglevel=DEBUG
sudo snap set edgex-app-rfid-llrp-inventory config.service-startupmsg="#test start up message#"
sudo snap start edgex-app-rfid-llrp-inventory
```

3. Check the .env file is valid now without `export` command:
```
cat /var/snap/edgex-app-rfid-llrp-inventory/current/config/app-rfid-llrp-inventory/res/app-rfid-llrp-inventory.env 

SERVICE_STARTUPMSG="#test start up message#"
WRITABLE_LOGLEVEL="DEBUG"
```
4. Validate env vars in `.env` have been exported successfully:
```
sudo snap logs edgex-app-rfid-llrp-inventory

msg="#test start up message#"
level=DEBUG msg="xxx"
```
5. Validate autostart via snap option works:
```
$ sudo snap set edgex-app-rfid-llrp-inventory autostart=true
$ snap services edgex-app-rfid-llrp-inventory
Service                                                Startup  Current  Notes
edgex-app-rfid-llrp-inventory.app-rfid-llrp-inventory  enabled  active   -

$ sudo snap set edgex-app-rfid-llrp-inventory autostart=false
$ snap services edgex-app-rfid-llrp-inventory
Service                                                Startup   Current   Notes
edgex-app-rfid-llrp-inventory.app-rfid-llrp-inventory  disabled  inactive  -
```

### CODE REVIEW

**Comment by farshidtz** [comment]
@MonicaisHer please complete the PR template.

**Comment by MonicaisHer** [comment]
> @MonicaisHer please complete the PR template (description, checklist).

Thanks, updated.

File: snap/local/helper-go/Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+clean:
+	rm -f helper-go
+
+# TODO: fixup unit testing
+test:
+	$(GO) test -coverprofile=coverage.out ./...
+	$(GO) vet ./...
+	gofmt -l .
+	[ "`gofmt -l .`" = "" ]
```
Comment (farshidtz) [inline snap/local/helper-go/Makefile:16]: Can be removed. There are no tests and very little code to do vetting.

File: snap/hooks/install [hunk 2]
```diff
@@ -1,20 +1,3 @@
...
-# into $SNAP_DATA/config
-mkdir -p "$SNAP_DATA/config"
-if [ ! -f "$SNAP_DATA/config/app-rfid-llrp-inventory/res/configuration.toml" ]; then
-    mkdir -p "$SNAP_DATA/config/app-rfid-llrp-inventory/res"
-    cp "$SNAP/config/app-rfid-llrp-inventory/res/configuration.toml" "$SNAP_DATA/config/app-rfid-llrp-inventory/res/configuration.toml"
-    # do replacement of the $SNAP, $SNAP_DATA, $SNAP_COMMON environment variables in the config files
-    sed -i -e "s@\$SNAP_COMMON@$SNAP_COMMON@g" \
-        -e "s@\$SNAP_DATA@$SNAP_DATA_CURRENT@g" \
-        -e "s@\$SNAP@$SNAP_CURRENT@g" \
-        "$SNAP_DATA/config/app-rfid-llrp-inventory/res/configuration.toml"
```
Comment (farshidtz) [inline snap/hooks/install:19]: This code is simply removed because it wasn't doing anything?

File: snap/local/bin/source-env-file.sh [hunk 3]
```diff
@@ -17,7 +17,10 @@ logger "edgex service override: : SERVICE_ENV=$SERVICE_ENV"
 
 if [ -f "$SERVICE_ENV" ]; then
     logger "edgex service override: : sourcing $SERVICE_ENV"
-    source "$SERVICE_ENV"
+    set -o allexport
```
Comment (farshidtz) [inline snap/local/bin/source-env-file.sh:20]: Please update the logger to have the app name in prefix.
Also, rename SERVICE_ENV to something more meaningful.

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: snap/hooks/install (hunk 2 above)
Comment (MonicaisHer) [reply snap/hooks/install:19]: yes, installing all the config files from $SNAP to $SNAP_DATA has been done by the `hooks.CopyDir`.

**Review by farshidtz — APPROVED** [review]
Thanks. Looks good.

**Pull request merged**